libraries/botbuilder/src/botFrameworkAdapter.ts

```diff
@@ -2011,8 +2012,15 @@ export class BotFrameworkAdapter
     }
 
     private async readRequestBodyAsString(request: IReceiveRequest): Promise<Activity> {
-        const contentStream = request.streams[0];
-        return await contentStream.readAsJson<Activity>();
+        const [activityStream, ...attachmentStreams] = request.streams;
+
+        const activity = await activityStream.readAsJson<Activity>();
+
+        activity.attachments = await Promise.all(
```

Comment:
You probably are aware of this, but adding this note just in case -- attachments have a content type, and depending on that content type they could be json, cards, audio, text, etc. Another dimension is whether the attachment is inlined or has a url. I'd suggest to try this out with a few different contents and mechanism combinations to make sure we are covering the scenario families.

Reply:
If you know for a fact that this code is independent of any corner case combination then that's fine. In C# there are 2 families of attachments to test to cover pretty much all cases, I'm just not sure what the minimum test base is for this code without spending a bit more time on it.

Reply:
I am not expert on this topic, you can refer to how C# SDK works.

FYI, someone told me, stream could be never-ending, e.g. a speech stream. That means `await Promise.all()` could (almost) never resolve.

Maybe we didn't go as far as this yet even in C#.

Reply:
@compulim, the `IReceiveRequest`, in this context, is a discrete set of bytes from a socket. The socket, itself, is a stream that is never-ending.

@carlosscastro working on verifying now, thanks for the pointers! I think I have found another hiccup when adding `contentType`-aware stream reading.

Reply:
@compulim good point! the endless stream could exist in the highest protocol layer, which assembles streams. However, this is a requirement that was part of the original design but it's not completely implemented, and we don't really use it for any practical scenario and many parts of the code don't really support that.